### PR TITLE
Add image gallery. Add version to feed.

### PR DESCRIPTION
**Pull request by Jamstaz**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+### Salesfire v1.2.4
+Released on 2020-02-10
+Release notes:
+
+- Add product image fallback
+- Add verison to feed
```
Comment (StephenHonor) [inline CHANGELOG.md:8]: Fixed stock level

File: Cron/Feed.php [hunk 2]
```diff
@@ -303,15 +311,27 @@ public function execute()
...
+                            $this->printLine($siteId, '<stock>' . ($stock_qty ? $stock_qty : 0) .'</stock>', 5);
+                        }
 
                         $this->printLine($siteId, '<link>' . $product->getProductUrl(true) . '</link>', 5);
 
                         $image = $product->getImage();
-                        if (! empty($image)) {
-                            $this->printLine($siteId, '<image>' . $mediaUrl . 'catalog/product' . $image . '</image>', 5);
+                        if (empty($image) || $image == 'no_selection') {
+                            $image = $product->getMediaGalleryImages()->getFirstItem()->getUrl();
```
Comment (StephenHonor) [inline Cron/Feed.php]: Does this need to be done on the other bit as well? Also is it worth putting in a separate function like magento 1?

File: Cron/Feed.php [hunk 3]
```diff
@@ -132,6 +132,9 @@ public function execute()
                 $products = $this->getVisibleProducts($storeId, $page);
                 $count = count($products);
 
+                $object_manager = \Magento\Framework\App\ObjectManager::getInstance();
+                $stock_state = $object_manager->get('\Magento\CatalogInventory\Api\StockStateInterface');
```
Comment (StephenHonor) [inline Cron/Feed.php]: Does this need to be defined on each loop?

File: Cron/Feed.php [hunk 4]
```diff
@@ -247,8 +250,13 @@ public function execute()
 
                                 $this->printLine($siteId, '<mpn><![CDATA['.$this->escapeString($childProduct->getSku()).']]></mpn>', 5);
 
-                                $stock_item = $this->_stockItem->get($childProduct->getId());
-                                $this->printLine($siteId, '<stock>'.($stock_item && $stock_item->getIsInStock() ? ($stock_item->getQty() > 0 ? (int) $stock_item->getQty() : 1) : 0).'</stock>', 5);
+                                try {
+                                    $stock_item = $this->_stockItem->get($childProduct->getId());
```
Comment (StephenHonor) [inline Cron/Feed.php:256]: Can you change this to set the stock level as a variable and only print the line once please?

File: Cron/Feed.php [hunk 5]
```diff
@@ -303,15 +311,27 @@ public function execute()
                         }
 
                         $this->printLine($siteId, '<mpn><![CDATA['.$this->escapeString($product->getSku()).']]></mpn>', 5);
-
-                        $stock_item = $this->_stockItem->get($product->getId());
-                        $this->printLine($siteId, '<stock>'.($stock_item && $stock_item->getIsInStock() ? ($stock_item->getMinQty() > 0 ? (int) $stock_item->getQty() : 1) : 0).'</stock>', 5);
+                        $stock_item = null;
+
+                        try {
+                            $stock_item = $this->_stockItem->get($product->getId());
```
Comment (StephenHonor) [inline Cron/Feed.php:315]: Same as previous

**Review by StephenHonor — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Jamstaz) [reply CHANGELOG.md:8]: Added.

File: Cron/Feed.php (hunk 2 above)
Comment (Jamstaz) [reply Cron/Feed.php]: Yeah, I've added a separate function.

File: Cron/Feed.php (hunk 3 above)
Comment (Jamstaz) [reply Cron/Feed.php]: Nope.

File: Cron/Feed.php (hunk 4 above)
Comment (Jamstaz) [reply Cron/Feed.php:256]: Yes

File: Cron/Feed.php (hunk 5 above)
Comment (Jamstaz) [reply Cron/Feed.php:315]: Done

**Review by StephenHonor — APPROVED** [review]

**Pull request merged**